core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -767,6 +784,9 @@ public MergingCompletableFuture(InvocationContext ctx, int participants, T[] res
 
       @Override
       protected Object result() {
+         if (hasUnsureResponse && lostData) {
```

Comment:
No, this is correct; If we've lost data it shouldn't throw but return limited stream. If we've got unsure response - hakuna matata, there has been another response delivering the results. Only if those two combine we'll rather retry.